frontend/src/Initialize.tsx

```diff
@@ -219,22 +219,25 @@ const Initialize = (props: Props) => {
 
   const saveGameSettings = (evt: any) => {
     evt.preventDefault();
-    localStorage.setItem('gameSettingsInLocalStorage', JSON.stringify(props.state.propagated));
+    localStorage.setItem(
```

Comment:
we might want to use the abstractions here: https://github.com/rbtying/shengji/blob/08b6f8c89213ce4b11def917b6f4f191a3f060c2/frontend/src/localStorageState.ts#L26 

though I think in practice it's basically the same?

Reply:
why does persist has a before param that is never used?

this looks strange:

    stringLocalStorageState("gameSettingsInLocalStorage", '').persist(null, JSON.stringify(props.state.propagated));

  much simpler using localStorage directly.

Reply:
also, to use the instance generated by stringLocalStorageState for retrieval, a const needs to be created every time Initialize is called, not that efficient. I will stick with using localStorage directly for now.

Reply:
hm. the intent with the abstraction is to keep things type-safe -- the `stringLocalStorageState` wraps a serializer and deserializer from the underlying localstorage, which you are using JSON.stringify / etc for.

I can refactor this later if it's confusing -- not a big deal for now.